### ISSUE TITLE
EE10 `ServletRequest.isSecure()` not set by `ForwardedRequestCustomizer`

### ISSUE THREAD

**Issue by rstar2** [eclipse/jetty.project#10771]
**Jetty version(s)**
12.0.2

**Jetty Environment**
ee10

**Java version/vendor** `(use: java -version)`
17

**OS type/version**
Linux

**Description**
Outcome of the http-forwarded.mod  in Jetty 12 is different that  from Jetty 11/10/....

When the `http-forwarded.mod` is on and a proxy serveris is sending the header `X-Forwarded-Proto: https` then the ForwardedRequestCustomizer.java is not making the ServletRequest.isSecure() return false as it was doing in previous versions. 
I'll point to code snippets from the source code for easier explanation:
In the ForwardedRequestCustomizer.customize() method the "secure" flag is properly detected from the X-Forwarded-Proto header and in it is used in the in the returned in 

``` java
 return new Request.Wrapper(request)     {

            @Override
            public HttpURI getHttpURI()
            {
                return uri;
            }

            @Override
            public HttpFields getHeaders()
            {
                return headers;
            }

            @Override
            public boolean isSecure()
            {
                return secure || super.isSecure();
            }

            @Override
            public ConnectionMetaData getConnectionMetaData()
            {
                return connectionMetaData;
            }
        };
```

But the org.eclipse.jetty.ee10.servlet.ServletApiRequest.isSecure() is calling the
``` java
   @Override
    public boolean isSecure()
    {
        return getRequest().getConnectionMetaData().isSecure();
    }
```

And the overriden method from the customizer is actually `getRequest().isSecure()`.

Also the other similar methods like Servlet.getRemoteAddr() are handled differently (correctly) like:
``` java
@Override
    public String getRemoteAddr()
    {
        return Request.getRemoteAddr(getRequest());
    }
```

**Comment by joakime** [comment]
Bug only exists on EE10, not EE9/EE8.

**Comment by joakime** [comment]
Opened PR #10772

**Comment by sbordet** [comment]
Fixed by #10772.
